lib/streamlit/errors.py

```diff
@@ -376,21 +376,23 @@ def __init__(self):
 class StreamlitPageNotFoundError(LocalizableStreamlitException):
     """Exception raised the linked page can not be found."""
 
-    def __init__(self, page: str, main_script_directory: str, is_mpa_v2: bool):
+    def __init__(self, page: str, main_script_directory: str):
         directory = os.path.basename(main_script_directory)
 
         message = (
-            "Could not find page: `{page}`. You must provide a file path "
-            "relative to the entrypoint file (from the directory `{directory}`). "
-            "Only the entrypoint file and files in the `pages/` directory are supported."
+            "Could not find page: `{page}`. You must provide a `StreamlitPage` "
+            "object or file path relative to the entrypoint file. Only pages "
+            "previously defined by `st.Page` and passed to `st.navigation` are "
+            "allowed."
         )
 
-        if is_mpa_v2:
+        from streamlit.runtime.pages_manager import PagesManager
```

Comment:
nit: I would move this import to the top of the file. In my opinion, we should only do these in-function imports for very good reasons, such as the import is slow and, thus, we want to lazy import.

Reply:
Good point. Updating!

Reply:
Ah. I updated it, and we get some circular import issues, so I unfortunately have to keep this import here (at least it's a good reason!).

Reply:
oh I see! I don't think that a circular import is a good reason because it hints at a flaw in our code design. But I believe historically files like `errors.py` has been problematic because they are used under-the-hood from elements etc. which makes it difficult to get away from this in all circumstances. 
In this case here, I think we can avoid the in-function import and would prefer if we passed the information from the caller [here](https://github.com/streamlit/streamlit/pull/10458/files#diff-5fe41e0bd5c4553e4c24998cc7cd227dc719b689a9fd0a83777e29a1c2690d02R887) instead of doing the import. It looks like there we have access to the `pages_manager` and the `PageNotFound` error could accept the `uses_pages_directory` flag as an argument instead.
What do you think?